SYCL/Matrix/element_wise_all_ops_half.cpp

```diff
@@ -69,7 +69,7 @@ void matrix_verify_add(queue q, big_matrix<T, M, N> &A, nd_range<2> &r,
 
            auto wi_slice_a = sub_a.get_wi_data();
            for (int i = 0; i < wi_slice_a.length(); i++) {
-             wi_slice_a[i] = wi_slice_a[i] + 2;
+             wi_slice_a[i] = wi_slice_a[i] + static_cast<half>(2);
```

Comment:
static_cast should be on wi_slice_a[i], since wi_elem<half> can't be converted to double implicitly.
wi_slice_a[i] = static_cast<half>(wi_slice_a[i]) + 2

Reply:
The compilation and runtime gives me same results for both:
wi_slice_a[i] = wi_slice_a[i] + static_cast<half>(2);
and
wi_slice_a[i] = static_cast<half>(wi_slice_a[i]) + 2;
For other types, we don't need static_cast on wi_slice[i], right? why we need it now for half.
static_cast<T>(RHS) should be taken care of in the code.